File: src/utils/resolve_truth_path.py
"""Utilities for locating the canonical truth file.

This mirrors ``MATLAB/src/utils/resolve_truth_path.m`` and returns the
preferred path for the truth file if it exists.
"""

from __future__ import annotations

from pathlib import Path


def resolve_truth_path() -> str | None:
    """Resolve the path to the truth file.

    Returns the canonical truth file path if found, otherwise ``None``.
    """

    preferred = Path(__file__).resolve().parents[2] / "DATA" / "Truth" / "STATE_X001.txt"

    if preferred.is_file():
        print(f"Using TRUTH: {preferred}")
        return str(preferred)

    return None

File: MATLAB/src/utils/resolve_truth_path.m
function truth_path = resolve_truth_path()
%RESOLVE_TRUTH_PATH Canonicalize the truth file path.
%   truth_path = resolve_truth_path()
%       Returns the path string to the truth file if found, or an empty char
%       if it does not exist.

    % Prefer repository DATA/Truth path relative to this file
    here = fileparts(mfilename('fullpath'));
    repo = fileparts(fileparts(here)); % .../MATLAB/src/utils -> repo
    preferred = fullfile(repo,'DATA','Truth','STATE_X001.txt');

    if isfile(preferred)
        truth_path = preferred;
        fprintf('Using TRUTH: %s\n', truth_path);
        return;
    end

    truth_path = '';
end
